### PR TITLE
build(replay-tool): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to node16 module resolution

### CODE REVIEW

File: packages/tools/replay-tool/package.json [hunk 1]
```diff
@@ -32,7 +43,7 @@
 		"lint:fix": "npm run prettier:fix && npm run eslint:fix",
 		"prettier": "prettier --check . --cache --ignore-path ../../../.prettierignore",
 		"prettier:fix": "prettier --write . --cache --ignore-path ../../../.prettierignore",
-		"tsc": "tsc"
+		"tsc": "fluid-tsc commonjs --project ./tsconfig.cjs.json && copyfiles -f ../../../common/build/build-common/src/cjs/package.json ./dist"
```
Comment (DLehenbauer) [inline packages/tools/replay-tool/package.json:46]: Probably an easy conversion to ESM if downstream snapshot tests is converted.

Comment (jason-ha) [reply packages/tools/replay-tool/package.json:48]: Can we convert both to ESM only at same time?

File: packages/tools/replay-tool/package.json [hunk 2]
```diff
@@ -12,18 +12,29 @@
...
 	"bin": {
 		"replayTool": "bin/replayTool"
 	},
 	"scripts": {
-		"api": "fluid-build . --task api",
-		"api-extractor:commonjs": "api-extractor run --local",
 		"build": "fluid-build . --task build",
 		"build:compile": "fluid-build . --task compile",
-		"build:docs": "fluid-build . --task api",
-		"ci:build:docs": "api-extractor run",
```
[the remark this answers is not in the document]
Comment (tylerbutler) [reply packages/tools/replay-tool/package.json:28]: I don't have a strong preference, but unless the types have doc comments already, there's not much point to generating api docs anyway.

File: packages/tools/replay-tool/src/index.ts [hunk 3]
```diff
@@ -8,6 +8,6 @@ export {
 	getNormalizedFileSnapshot,
 	loadContainer,
 	uploadSummary,
-} from "./helpers";
-export { ReplayArgs } from "./replayArgs";
-export { ReplayTool } from "./replayMessages";
+} from "./helpers.js";
```
Comment (tylerbutler) [inline packages/tools/replay-tool/src/index.ts:11]: Maybe out of scope for this PR but are all of these exports really needed?

Comment (sonalideshpandemsft) [reply packages/tools/replay-tool/src/index.ts:11]: just checked - all the exports are used.

Comment (tylerbutler) [reply packages/tools/replay-tool/src/index.ts:11]: Thanks for the due diligence!

**Review by tylerbutler — APPROVED** [review]

**Pull request merged**